crawl-ref/source/spl-summoning.cc

```diff
@@ -2543,61 +2439,22 @@ bool aim_battlesphere(actor* agent, spell_type spell, int powc, bolt& beam)
         // target-seeking action, cancel it so that it can focus on a new
         // target
         reset_battlesphere(battlesphere);
-
-        // Don't try to fire at ourselves
-        if (beam.target == battlesphere->pos())
-            return false;
-
-        // If the player beam is targeted at a creature, aim at this creature.
-        // Otherwise, aim at the furthest creature in the player beam path
-        bolt testbeam = beam;
-
-        if (agent->is_player())
-            testbeam.thrower = KILL_YOU_MISSILE;
-        else
-        {
-            testbeam.thrower = KILL_MON_MISSILE;
-            testbeam.source_id = agent->mid;
-        }
-
-        testbeam.is_tracer = true;
-        zap_type ztype = spell_to_zap(spell);
-
-        // Fallback for non-standard spell zaps
-        if (ztype == NUM_ZAPS)
-            ztype = ZAP_MAGIC_DART;
-
-        // This is so that reflection and pathing rules for the parent beam
-        // will be obeyed when figuring out what is being aimed at
-        zappy(ztype, powc, false, testbeam);
-
-        battlesphere->props["firing_target"] = beam.target;
         battlesphere->props.erase("foe");
-        if (!actor_at(beam.target))
-        {
-            testbeam.fire();
 
-            for (const coord_def c : testbeam.path_taken)
+        // Pick a random baddie in LOS
```

Comment:
I'm not sure about this change at all, why is it bad to target whatever the main spell is aiming at, regardless of how that aim is achieved? Randomness just seems to add player lack-of-control here, and I don't see the gain

Reply:
I went with a consistent targeting scheme across all spells instead of the special-case different targeting rules for different spells for clarity of effect. 

The old targeting scheme was (roughly):

- If aimed at a target hit that,
- Else if there's a target in the beam path of the aim hit that,
- Else if the mirrored spell is an explosion hit something it explodes on

And I don't see the gain of adding an "else if the mirrored spell is one of the positional spells hit something that the spell hit" compared to simplifying the targeting scheme to always shoot something.

Reply:
We had some back-and-forth in IRC about improved battlesphere targeting; I'm not convinced its worth the effort and complexity, but I made the simplification commit separate for an easy revert.